conkit/io/mapalign.py

```diff
@@ -0,0 +1,118 @@
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+"""
+Parser module specific to map_align map files
+"""
+
+from conkit.io._parser import ContactFileParser
+from conkit.core.contact import Contact
+from conkit.core.contactmap import ContactMap
+from conkit.core.contactfile import ContactFile
+
+
+class MapAlignParser(ContactFileParser):
+    """Class to parse a map_align map file
+    """
+
+    def read(self, f_handle, f_id="map_align"):
+        """Read a contact file
+
+        Parameters
+        ----------
+        f_handle
+           Open file handle [read permissions]
+        f_id : str, optional
+           Unique contact file identifier
+
+        Returns
+        -------
+        :obj:`~conkit.core.contactfile.ContactFile`
+
+        Raises
+        ------
+        :exc:`RuntimeError`
+           Sequence length exceeds the length stated with the LEN card
+
+        """
+
+        hierarchy = ContactFile(f_id)
+        _map = ContactMap("map_1")
+        hierarchy.add(_map)
+        seq_length = None
+
+        for line in f_handle:
+            line = line.strip().split()
+
+            if line[0] == "LEN":
+                seq_length = int(line[1])
+                continue
+
+            elif line[0] != "CON":
+                continue
+
+            elif line[1].isdigit() and line[2].isdigit():
+                line.pop(0)
+                if seq_length is not None and (int(line[0]) > seq_length or int(line[1]) > seq_length):
+                    raise RuntimeError(
+                        "Contacting residues outside sequence range ({} residues)! Conflicting contact: {}-{}".format(
+                            seq_length, line[0], line[1]))
+                _contact = Contact(int(line[0]), int(line[1]), float(line[2]))
+                _map.add(_contact)
+
+        hierarchy.method = "Contact map compatible with map_algin"
+
+        return hierarchy
+
+    def write(self, f_handle, hierarchy):
+        """Write a contact file instance to to file
+
+        Parameters
+        ----------
+        f_handle
+           Open file handle [write permissions]
+        hierarchy : :obj:`~conkit.core.contactfile.ContactFile`, :obj:`~conkit.core.contactmap.ContactMap`
+                    or :obj:`~conkit.core.contact.Contact`
+
+        Raises
+        ------
+        :exc:`RuntimeError`
+           More than one contact map in the hierarchy
+
+        """
+        contact_file = self._reconstruct(hierarchy)
+        if len(contact_file) > 1:
+            raise RuntimeError("More than one contact map provided")
+        seq_length = max([max(contact.res1_seq, contact.res2_seq) for contact in contact_file.top_map])
+        content = "LEN {}\n".format(seq_length)
+        for contact_map in contact_file:
```

Comment:
this loop is kinda redundant. I get why it's there but not really necessary since `>1` is not supported.